src/types/component.go

```diff
@@ -21,50 +21,93 @@ type ZarfComponent struct {
 	// Default changes the default option when deploying this component
 	Default bool `json:"default,omitempty" jsonschema:"description=Determines the default Y/N state for installing this component on package deploy"`
 
-	// Required makes this component mandatory for package deployment
-	Required bool `json:"required,omitempty" jsonschema:"description=Do not prompt user to install this component, always install on package deploy"`
+	// DeprecatedRequired makes this component mandatory for package deployment
+	DeprecatedRequired *bool `json:"required,omitempty" jsonschema:"description=[Deprecated] Do not prompt user to install this component, always install on package deploy. This will be removed in Zarf v1.0.0.,deprecated=true"`
```

Comment:
I'm sure this is just a side effect on the split, but I think we'll want to wait until optional is introduced before we mark required as deprecated

Reply:
This is why this is marked as a breaking change. The changeover is happening at the same time the deprecation is being marked.